lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -77,6 +77,8 @@ sub _git_clone ($app, $job, $ctx, $path, $url) {
         $url = $git->get_origin_url;
     }
 
+    return if ($requested_branch and $requested_branch !~ tr/a-f0-9//c and $git->check_sha($requested_branch));
```

Comment:
Just don't create branches named `coffee` or `beef` ;-)

Reply:
I just saw an example `https://github.com/rbmarliere/os-autoinst-distri-opensuse.git#162791` on o3.
So it's probably better to doublecheck the output of rev-parse. if it returns the same string, it's a sha, otherwise it's a branch name.

Reply:
I guess we can live with that limitation - although putting in a regex that would not only check the characters but also the length probably wouldn't be too much of an effort.

Reply:
If by limitation you mean breaking branch names that could be sha's I vote against it. That would be very confusing in practice.

Reply:
I haven't seen @perlpunk 's last message when I wrote mine. So you can disregard my last message.

Reply:
I'm now checking if the given sha/branch matches the beginning of the returned sha from rev-parse